pkgs/development/libraries/microsoft_gsl/default.nix

```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchgit, cmake }:
+
+stdenv.mkDerivation rec {
+  name = "microsoft_gsl-${version}";
+  version = "2017-02-13";
+
+  src = fetchgit {
+    url = "https://github.com/Microsoft/GSL.git";
+    rev = "3819df6e378ffccf0e29465afe99c3b324c2aa70";
+    sha256 = "03d17mnx6n175aakin313308q14wzvaa9pd0m1yfk6ckhha4qf35";
```

Comment:
I used `fetchgit` instead of `fetchFromGithub` as I need the submodules to be checked out in order to run the test suite.